### ISSUE TITLE
artifacts offline install - implausibly old time stamp 1970-01-01 00:00:00

### ISSUE THREAD

**Issue by fgelcer** [scylladb/scylladb#13415]
## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

*Describe your issue in detail and steps it took to produce it.*

## Impact

installation of ScyllaDB on a fresh CentOS 7 failed (it may actually succeeded, but the test stopped because of the errors encountered during the installation).

```
2023-03-31 05:56:39.295: (TestFrameworkEvent Severity.ERROR) period_type=one-time event_id=507e9bf6-1e7d-4688-8718-7032e7d0c50a, source=ArtifactsTest.SetUp()
exception=[Node artifacts-centos7-jenkins-db-node-804b2316-0-1 [104.196.128.187 | 10.142.0.39] (seed: True)] NodeSetupFailed: Encountered a bad command exit code!
Command: 'sudo bash -cxe "\ntar xvfz ./unified_package.tar.gz\ncd ./scylla-*\n./install.sh --housekeeping\ncd -\nrm -f /tmp/scylla.yaml\n"'
Exit code: 1
Stdout:
fs.nr_open = 1073741816
fs.inotify.max_user_instances = 1200
kernel.sched_tunable_scaling = 0
kernel.sched_min_granularity_ns = 500000
kernel.sched_wakeup_granularity_ns = 450000
kernel.sched_latency_ns = 1000000
kernel.sched_autogroup_enabled = 0
kernel.numa_balancing = 0
vm.swappiness = 1
Scylla offline install completed.
Stderr:
+ tar xvfz ./unified_package.tar.gz
tar: scylla-5.3.0~dev/scylla-python3/bin/python3.11: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/python3: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/geomet: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/traceback-with-variables: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/scylla-api-client: implausibly old time stamp 1970-01-01 00:00:00
+ cd ./scylla-5.3.0~dev
+ ./install.sh --housekeeping
Traceback (most recent call last):
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/cluster.py", line 3640, in node_setup
cl_inst.node_setup(_node, **setup_kwargs)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/cluster.py", line 4291, in node_setup
self._scylla_install(node)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/cluster.py", line 4370, in _scylla_install
node.offline_install_scylla(unified_package=self.params.get('unified_package'),
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/cluster.py", line 2062, in offline_install_scylla
self.remoter.run('sudo bash -cxe "%s"' % install_cmds)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/remote/remote_base.py", line 613, in run
result = _run()
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/utils/decorators.py", line 67, in inner
return func(*args, **kwargs)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/remote/remote_base.py", line 604, in _run
return self._run_execute(cmd, timeout, ignore_status, verbose, new_session, watchers)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/remote/remote_base.py", line 537, in _run_execute
result = connection.run(**command_kwargs)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/remote/libssh2_client/__init__.py", line 620, in run
return self._complete_run(channel, exception, timeout_reached, timeout, result, warn, stdout, stderr)
File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/remote/libssh2_client/__init__.py", line 654, in _complete_run
raise UnexpectedExit(result)
sdcm.remote.libssh2_client.exceptions.UnexpectedExit: Encountered a bad command exit code!
Command: 'sudo bash -cxe "\ntar xvfz ./unified_package.tar.gz\ncd ./scylla-*\n./install.sh --housekeeping\ncd -\nrm -f /tmp/scylla.yaml\n"'
Exit code: 1
Stdout:
fs.nr_open = 1073741816
fs.inotify.max_user_instances = 1200
kernel.sched_tunable_scaling = 0
kernel.sched_min_granularity_ns = 500000
kernel.sched_wakeup_granularity_ns = 450000
kernel.sched_latency_ns = 1000000
kernel.sched_autogroup_enabled = 0
kernel.numa_balancing = 0
vm.swappiness = 1
Scylla offline install completed.
Stderr:
+ tar xvfz ./unified_package.tar.gz
tar: scylla-5.3.0~dev/scylla-python3/bin/python3.11: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/python3: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/geomet: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/traceback-with-variables: implausibly old time stamp 1970-01-01 00:00:00
tar: scylla-5.3.0~dev/scylla-python3/bin/scylla-api-client: implausibly old time stamp 1970-01-01 00:00:00
+ cd ./scylla-5.3.0~dev
+ ./install.sh --housekeeping
```

## How frequently does it reproduce?

last 2 runs failed with this error

## Installation details

Cluster size: 1 nodes (n1-standard-2)

Scylla Nodes used in this run:
- artifacts-centos7-jenkins-db-node-804b2316-0-1 (104.196.128.187 | 10.142.0.39) (shards: 2)

OS / Image: `https://www.googleapis.com/compute/v1/projects/centos-cloud/global/images/family/centos-7` (gce: us-east1)

Test: `artifacts-centos7-test`
Test id: `804b2316-eb68-46f6-a2a5-f20aba954834`
Test name: `scylla-master/artifacts-offline-install/artifacts-centos7-test`
Test config file(s):

- [centos7.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/a317398419d94861af4e5fdf7b945c7655d4cc77/test-cases/artifacts/centos7.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 804b2316-eb68-46f6-a2a5-f20aba954834`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=804b2316-eb68-46f6-a2a5-f20aba954834)
- Show all stored logs command: `$ hydra investigate show-logs 804b2316-eb68-46f6-a2a5-f20aba954834`

## Logs:
- **db-cluster-804b2316.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/804b2316-eb68-46f6-a2a5-f20aba954834/20230331_055758/db-cluster-804b2316.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/804b2316-eb68-46f6-a2a5-f20aba954834/20230331_055758/db-cluster-804b2316.tar.gz)
- **sct-runner-events-804b2316.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/804b2316-eb68-46f6-a2a5-f20aba954834/20230331_055758/sct-runner-events-804b2316.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/804b2316-eb68-46f6-a2a5-f20aba954834/20230331_055758/sct-runner-events-804b2316.tar.gz)
- **sct-804b2316.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/804b2316-eb68-46f6-a2a5-f20aba954834/20230331_055758/sct-804b2316.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/804b2316-eb68-46f6-a2a5-f20aba954834/20230331_055758/sct-804b2316.log.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-centos7-test/244/)
</details>

**Comment by fgelcer** [comment]
@yaronkaikov

**Comment by yaronkaikov** [comment]
@syuu1228 Can you please have a look?

**Comment by syuu1228** [comment]
The unified package url was https://downloads.scylladb.com/unstable/scylla/master/relocatable/2023-03-31T03:19:03Z/scylla-unified-5.3.0~dev-0.20230331.160c184d0b0f.x86_64.tar.gz, I checked timestamp of the tar file, it contains incorrect timestamp files as reported:
```
$ tar tvpf scylla-unified-5.3.0~dev-0.20230331.160c184d0b0f.x86_64.tar.gz |grep scylla-python3/bin
drwxr-xr-x jenkins/jenkins        0 2023-03-31 10:25 scylla-5.3.0~dev/scylla-python3/bin/
-rwxr-xr-x jenkins/jenkins      275 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/python3.11
lrwxrwxrwx jenkins/jenkins        0 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/python3 -> python3.11
-rwxr-xr-x jenkins/jenkins      212 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/geomet
-rwxr-xr-x jenkins/jenkins     1047 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/traceback-with-variables
-rwxr-xr-x jenkins/jenkins      229 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/scylla-api-client
``` 


I also checked locally built unified package, the result was same:
```
$ tar tvpf build/release/dist/tar/scylla-unified-5.3.0~dev-0.20230402.f2cef98e7c13.x86_64.tar.gz |grep scylla-python3/bin
drwxr-xr-x syuu/docker        0 2023-04-02 18:25 scylla-5.3.0~dev/scylla-python3/bin/
-rwxr-xr-x syuu/docker      275 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/python3.11
lrwxrwxrwx syuu/docker        0 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/python3 -> python3.11
-rwxr-xr-x syuu/docker      212 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/geomet
-rwxr-xr-x syuu/docker     1047 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/traceback-with-variables
-rwxr-xr-x syuu/docker      229 1970-01-01 09:00 scylla-5.3.0~dev/scylla-python3/bin/scylla-api-client
$ tar tvpf build/release/dist/tar/scylla-python3-5.3.0~dev-0.20230402.f2cef98e7c13.x86_64.tar.gz |grep scylla-python3/bin
-rwxr-xr-x 0/0             275 1970-01-01 09:00 scylla-python3/bin/python3.11
lrw-r--r-- 0/0               0 1970-01-01 09:00 scylla-python3/bin/python3 -> python3.11
-rwxr-xr-x 0/0             212 1970-01-01 09:00 scylla-python3/bin/geomet
-rwxr-xr-x 0/0            1047 1970-01-01 09:00 scylla-python3/bin/traceback-with-variables
-rwxr-xr-x 0/0             229 1970-01-01 09:00 scylla-python3/bin/scylla-api-clien
```

Maybe this is related with create-relocatable-package.py in scylla-python3 repo, will take a look.

**Comment by syuu1228** [comment]
Probably because we haven't added `mtime` of TarInfo object:
```
    ti = tarfile.TarInfo(name=os.path.join("bin", pybin))
    ti.size = len(thunk)
    ti.mode = 0o755
    ar.reloc_addfile(ti, fileobj=io.BytesIO(thunk))
    
    ti = tarfile.TarInfo(name=os.path.join("bin", "python3"))
    ti.type = tarfile.SYMTYPE
    ti.linkname = pybin
    ar.reloc_addfile(ti)
```
https://github.com/scylladb/scylla-python3/blob/master/scripts/create-relocatable-package.py

I will send a patch.

**Comment by syuu1228** [comment]
Sent a patch at  https://github.com/scylladb/scylla-python3/pull/37

**Comment by denesb** [comment]
Looks like "Fixes" work cross-repos too. https://github.com/scylladb/scylla-python3/pull/37 had a `Fixes scylladb/scylladb#13415` and that resulted as this issue being closed as Done after I merged it.
Note that the update of `tools/python3` submodule was only queued to next yet, it is not yet in master at the time this writing.

**Comment by juliayakovlev** [comment]
@DoronArazii 
This issue was reproduced with 2023.1. Looks like this fix is missed there

## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

*Describe your issue in detail and steps it took to produce it.*

## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details








Cluster size: 1 nodes (i3.large)

Scylla Nodes used in this run:
- artifacts-oel76-jenkins-db-node-80ba2e34-1 (54.195.135.134 | 10.4.2.34) (shards: 2)


OS / Image: `ami-0d6a24fe35fdf5dc4` (aws: eu-west-1)

Test: `artifacts-oel76-test`
Test id: `80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1`
Test name: `enterprise-2023.1/artifacts-offline-install/artifacts-oel76-test`
Test config file(s):

- [oel76.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/7c9830422ce7f997f031e89ba1567f8115df9764/test-cases/artifacts/oel76.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1)
- Show all stored logs command: `$ hydra investigate show-logs 80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1`


## Logs:
- **db-cluster-80ba2e34.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1/20230906_100245/db-cluster-80ba2e34.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1/20230906_100245/db-cluster-80ba2e34.tar.gz)
- **sct-runner-events-80ba2e34.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1/20230906_100245/sct-runner-events-80ba2e34.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1/20230906_100245/sct-runner-events-80ba2e34.tar.gz)
- **sct-80ba2e34.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1/20230906_100245/sct-80ba2e34.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1/20230906_100245/sct-80ba2e34.log.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2023.1/job/artifacts-offline-install/job/artifacts-oel76-test/22/)
[Argus](https://argus.scylladb.com/test/1ff76a78-4fbd-4d93-a069-18b2200b1f75/runs?additionalRuns[]=80ba2e34-2e7e-4a3f-a371-2cd5009ad8e1)
</details>

**Comment by mykaul** [comment]
Most likely - lacked the 'backport ...' labels.

**Comment by juliayakovlev** [comment]
> Most likely - lacked the 'backport ...' labels.

@mykaul 
The issue is reproduced in 2023.1, enterprise. Do we need to open the issue in enterprise? Or add backport lable to 2023.1?

**Comment by mykaul** [comment]
I've added the labels for backport, I think it should go via this path.

**Comment by Annamikhlin** [comment]
also failed on enterprise: https://jenkins.scylladb.com/job/scylla-enterprise/job/artifacts-offline-install/job/artifacts-oel76-test/122/

@denesb @mykaul - isn't this fix already merged to enterprise?

**Comment by mykaul** [comment]
I think we need an update of https://github.com/scylladb/scylla-python3 submodule update into Scylla (Enterprise) , no?

**Comment by denesb** [comment]
```
09:46 $ git log --oneline -3
7d49a46 (HEAD, origin/enterprise) dist: Support FIPS mode
30b8fc2 (origin/master, origin/HEAD) create-relocatable-package.py: fix timestamp of executable files
d2f57dd dist: redhat: provide only a single version
```
The fix is in enterprise already.

**Comment by mykaul** [comment]
So we'll need to understand better why it failed then.

**Comment by avikivity** [comment]
Why is this being considered for backport? the timestamp of some files doesn't strike me as important.

**Comment by avikivity** [comment]
Backported to 5.2. 5.1 deemed unnecessary because few new installations are likely to use it (and installing from tarball is rare).

**Comment by Annamikhlin** [comment]
The issue seen on `2024.1` - https://jenkins.scylladb.com/job/enterprise-2024.1/job/artifacts-offline-install/job/artifacts-oel76-test/2/
and we still have it on enterprise - https://jenkins.scylladb.com/job/scylla-enterprise/job/artifacts-offline-install/job/artifacts-oel76-test/117/

**Comment by mykaul** [comment]
@avikivity - do we need a 5.4 backport for this, based on the comment by @Annamikhlin above?

**Comment by denesb** [comment]
> @avikivity - do we need a 5.4 backport for this, based on the comment by @Annamikhlin above?

This is already in 5.4 as scylladb/scylla-python3@30b8fc21bd2e8b32119e335a33aa3ae9fecaac03

**Comment by mykaul** [comment]
So it's not fixed, which is even more worrying :-/

**Comment by denesb** [comment]
> So it's not fixed, which is even more worrying :-/

I mean the fix is already in 5.4, 5.2 and 5.1. No more backporting is needed.
